### DISCUSSION TITLE
Could multi-tensor images be used for, for example, FLIM data?

### DISCUSSION THREAD

**Discussion by patta42** [DIPlib/diplib#180] [General]
I have read the docs of DIPimage and one thing particularly catched my attention: tensor images.  

I am looking for a way to store (and consequently, analyze) data in Matlab that is represented as an image, but every single pixel is the result of some more complex operation than just the light intensity measured. One example would be data from fluorescence lifetime imaging, where every pixels represents the result of a fit of a (in the most simple case) exponential decay to a histogram of the photon arrival times on the detector.

Another example is the height representation of a sample scanned by Scanning Ion Conductance Microscopy, where every pixel is the z-value of a piezo element at which point the current measured by the instrument exceed a selected threshold.

In both cases, every pixel has quite a lot of additional data (photon counts and arrival time or current and piezo deflection), and I'm looking for a way to store (and work with) this data.

I am not asking you to implement this ;-). But instead of starting from scratch (well, I have a quite large code base already) I'm interested in whether you think that diplib might be a suitable base to expand.

**Comment by crisluengo** [comment]
You can definitely use DIPimage images for this, but you should first think about how you want to interact with your data. DIPimage makes it easy to work with images where each pixel is a vector or a matrix, you can apply matrix operations on all pixels at once, and the indexing is quite simple. But I don’t know what type of operations you’d be applying to those images.

Note that in DIPimage 3 the images are stored in a single memory block, such that all values associated to one pixel are contiguous in memory. (In DIPimage 2 each tensor element (each channel) was a separate memory block.)
